azurelinuxagent/pa/rdma/suse.py

```diff
@@ -93,8 +93,7 @@ def install_driver(self):
                 msg = 'RDMA: Successfully installed "%s" from '
                 msg += 'configured repositories'
                 logger.info(msg % complete_name)
-                self.load_driver_module()
-                if requires_reboot:
+                if not self.load_driver_module() or requires_reboot:
```

Comment:
This will lead to an endless reboot loop and the user cannot even diagnose what is wrong. and no bug can be filed If for whatever reason the driver will not load.

Lets say we make a mistake and ens up with an API incompatibility in the package which would cause modprobe to fail. Rebooting the system is certainly not going to help the situation. On reboot we'll try again and then again on the next reboot.....